includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -0,0 +1,510 @@
+	 *
+	 * This only fires if can_register returns true.
+	 */
+	public function register() {
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_editor_assets' ] );
+	}
+
+	/**
+	 * Enqueue the editor scripts.
+	 */
+	public function enqueue_editor_assets() {
+		$settings = $this->get_settings();
+
+		// Don't load our custom post excerpt if excerpt functionality isn't turned on.
+		if ( ! isset( $settings['enable_excerpt'] ) || 1 !== (int) $settings['enable_excerpt'] ) {
+			return;
+		}
+
+		// This script removes the core excerpt panel and replaces it with our own.
+		wp_enqueue_script(
+			'classifai-post-excerpt',
+			CLASSIFAI_PLUGIN_URL . 'dist/post-excerpt.js',
+			get_asset_info( 'post-excerpt', 'dependencies' ),
+			get_asset_info( 'post-excerpt', 'version' ),
+			true
+		);
+	}
+
+	/**
+	 * Setup fields
+	 */
+	public function setup_fields_sections() {
+		$default_settings = $this->get_default_settings();
+
+		// Add the settings section.
+		add_settings_section(
+			$this->get_option_name(),
+			$this->provider_service_name,
+			function() {
+				printf(
+					wp_kses(
+						/* translators: %1$s is replaced with the OpenAI sign up URL */
+						__( 'Don\'t have an OpenAI account yet? <a title="Sign up for an OpenAI account" href="%1$s">Sign up for one</a> in order to get your API key.', 'classifai' ),
+						[
+							'a' => [
+								'href'  => [],
+								'title' => [],
+							],
+						]
+					),
+					esc_url( 'https://platform.openai.com/signup' )
+				);
+			},
+			$this->get_option_name()
+		);
+
+		// Add all our settings.
+		add_settings_field(
+			'api-key',
+			esc_html__( 'API Key', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'api_key',
+				'input_type'    => 'password',
+				'default_value' => $default_settings['api_key'],
+			]
+		);
+
+		add_settings_field(
+			'enable-excerpt',
+			esc_html__( 'Generate excerpt', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'enable_excerpt',
+				'input_type'    => 'checkbox',
+				'default_value' => $default_settings['enable_excerpt'],
+				'description'   => __( 'A button will be added to the excerpt panel that can be used to generate an excerpt.', 'classifai' ),
+			]
+		);
+
+		add_settings_field(
+			'length',
+			esc_html__( 'Excerpt length', 'classifai' ),
+			[ $this, 'render_input' ],
+			$this->get_option_name(),
+			$this->get_option_name(),
+			[
+				'label_for'     => 'length',
+				'input_type'    => 'number',
+				'max'           => 5,
+				'min'           => 1,
+				'step'          => 1,
+				'default_value' => $default_settings['length'],
+				'description'   => __( 'How many sentences should the excerpt be?', 'classifai' ),
+			]
+		);
+
+		add_settings_field(
+			'temperature',
+			esc_html__( 'Temperature value', 'classifai' ),
```

Comment:
I recognize that "Temperature value" and the "sampling temperature" and following helper text closely match what OpenAI is expecting, but I wonder if we might make this a bit easier to comprehend by making this less technical to something like "Excerpt precision" and "On a scale from 1 to 10 how precise do you want the excerpt to be?  10 is very focused and precise while 1 is less precise and random?" and then we convert that 1-10 value into the 0-2 increments?

Reply:
I debated even adding this setting in the first place. It's one of the options the [API supports](https://platform.openai.com/docs/api-reference/chat/create) but it's fairly confusing on what it does, not sure it's worth having. I guess I lean towards just removing this rather than trying to change this to make more sense and have to handle some sort of conversion behind the scenes

Reply:
Simpler by removal would be even better, maaaybe leave as a filter or just fully remove and we can look to add a filter later if there's a strong community desire?

Reply:
I've gone ahead and removed that. And there is already a filter around the request body we send, so this and any other API settings can easily be added/changed using that